runtime/src/main/java/org/corfudb/runtime/collections/CorfuQueue.java

```diff
@@ -0,0 +1,207 @@
+package org.corfudb.runtime.collections;
+
+import com.google.common.reflect.TypeToken;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.annotations.Accessor;
+import org.corfudb.annotations.ConflictParameter;
+import org.corfudb.annotations.CorfuObject;
+import org.corfudb.annotations.DontInstrument;
+import org.corfudb.annotations.Mutator;
+import org.corfudb.annotations.MutatorAccessor;
+import org.corfudb.runtime.view.OrderedGuidGenerator;
+
+import java.util.*;
+
+/**
+ * Persisted Queue supported by CorfuDB using distributed State Machine Replication.
+ * Entries enqueued are backed by a LinkedHashMap where each entry is mapped to a unique and
+ * globally ordered generated id that is returned upon successful <b>enqueue()</b>.
+ * <b>entryList()</b> returns the enqueued entries along with their ids sorted by
+ * the order in which their <b>enqueue()</b> operations materialized.
+ * Instead of a dequeue() this Queue supports a <b>remove()</b> which accepts the id of the element.
+ * Entries cannot be modified in-place and but can be removed from anywhere in the queue.
+ *
+ * Created by hisundar on 5/8/19.
+ *
+ * @param <E>   Type of the entry to be enqueued into the persisted queue
+ */
+@Slf4j
+@CorfuObject
+public class CorfuQueue<E> implements ISMRObject {
```

Comment:
No sure if `implements ISRObject` really needed here?

Reply:
Since CorfuQueue is just a wrapper is it still needed?
CorfuQueue is itself not a SMR object, but wraps one.

Reply:
done removed the ISMR